### PR TITLE
Regex'd away the -d suffix on doubles.

### PR DESCRIPTION
**Pull request by BenLand100**
Applied the seds in the following order in the data directory. This is compatible with jsoncpp and fastjson.
```bash
sed -i -E -e "s/([0-9]+\.[0-9]+)d([\+\-]*[0-9]+)/\1e\2/g" *.ratdb */*.ratdb
sed -i -E -e "s/([0-9]+\.[0-9]+e[\+\-]*[0-9]+)d/\1/g" *.ratdb */*.ratdb
sed -i -E -e "s/([0-9]+\.[0-9]+)d/\1/g" *.ratdb */*.ratdb
sed -i -E -e "s/([^x^0-9^A-F^a-f][0-9]+)d([\+\-]*[0-9]+)/\1e\2/g" *.ratdb */*.ratdb
sed -i -E -e "s/([^x^0-9^A-F^a-f][0-9]+)d/\1.0/g" *.ratdb */*.ratdb
sed -i -E -e "s/([0-9]+)\.d/\1.0/g" *.ratdb */*.ratdb
```

### CODE REVIEW

File: data/MATERIALS.ratdb [hunk 1]
```diff
@@ -512,13 +512,13 @@ name: "MATERIAL",
 index: "dodecane",
 valid_begin : [0, 0],
 valid_end : [0, 0],
-density: 0.749d,
+density: 0.749,
 nelements: 2,
 nmaterials: 0,
 elements: ["Carbon","Hydrogen"],
-elemprop: [.8563d, .1437d],
+elemprop: [.8563.0, .1437.0],
```
Comment (mastbaum) [inline data/MATERIALS.ratdb]: weird decimals!

File: data/MATERIALS.ratdb [hunk 2]
```diff
@@ -526,13 +526,13 @@ name: "MATERIAL",
 index: "ppo",
 valid_begin : [0, 0],
 valid_end : [0, 0],
-density: 1.06d,
+density: 1.06,
 nelements: 4,
 nmaterials: 0,
 elements: ["Carbon","Hydrogen","Nitrogen", "Oxygen",],
-elemprop: [.8142d, .0501d, .0633d, .0723d],
+elemprop: [.8142.0, .0501.0, .0633.0, .0723.0],
```
Comment (mastbaum) [inline data/MATERIALS.ratdb]: weird decimals!

File: data/MATERIALS.ratdb [hunk 3]
```diff
@@ -540,11 +540,11 @@ name: "MATERIAL",
 index: "gd_scintillator",
 valid_begin : [0, 0],
 valid_end : [0, 0],
-density: 0.846d,
+density: 0.846,
 nelements: 0,
 nmaterials: 4,
 materials: ["G4_Gd","mineral_oil","pseudocumene", "ppo",],
-matprop: [.00199646d, .798584d, 0.19765d, 0.00176991d],
+matprop: [.00199646.0, .798584.0, 0.19765, 0.00176991],
```
Comment (mastbaum) [inline data/MATERIALS.ratdb]: weird decimals!

File: data/MATERIALS.ratdb [hunk 4]
```diff
@@ -734,11 +734,11 @@ name: "MATERIAL",
 index: "polypropylene",
 valid_begin : [0, 0],
 valid_end : [0, 0],
-density: 0.855d, 
+density: 0.855, 
 nelements: 2,
 nmaterials: 0,
 elements: ["Hydrogen", "Carbon"],
-elemprop: [0.1437d, .8563d],  // chemical formula is C3H6, fractional weight same as polyethylene 
+elemprop: [0.1437, .8563.0],  // chemical formula is C3H6, fractional weight same as polyethylene
```
Comment (mastbaum) [inline data/MATERIALS.ratdb]: weird decimals!

**Comment by mastbaum** [comment]
Looks good except for this case with no leading zero and a `d` suffix, e.g.:

    .1337d -> .1337.0

**Comment by BenLand100** [comment]
Didn't think of the case where doubles started with a period... I'll check the snoplus commit for this in a bit.

**Comment by mastbaum** [comment]
Thanks! Say, did you re-`sed` and I actually caught them all, or fix the ones I found -- I definitely could have missed one.

**Comment by BenLand100** [comment]
A re-```sed``` was done just to be safe. 
> sed -i -E -e "s/(\.[0-9]+)\.0/0\1/g" *.ratdb */*.ratdb

**Comment by BenLand100** [comment]
Interestingly the parser managed to parse those numbers "correctly" but it's definitely invalid syntax  - I'll need to change that.

**Pull request merged**